pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -197,9 +200,151 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		}
 	}
 
+	//Begin AMQ resource reconcile
+	if instance.Spec.UseMessageBroker {
+		//check that address space exists
+		addressSpace := newAddressSpace(instance)
+
+		// Set UnifiedPushServer instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, addressSpace, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		foundAddressSpace := &enmassev1beta.AddressSpace{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: addressSpace.Name, Namespace: addressSpace.Namespace}, foundAddressSpace)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Address Space", "AddressSpace.Namespace", addressSpace.Namespace, "AddressSpace.Name", addressSpace.Name)
+			err = r.client.Create(context.TODO(), addressSpace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+		} else if err != nil {
+			return reconcile.Result{}, err
+		} else if !foundAddressSpace.Status.IsReady {
+			return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+		} else {
+			reqLogger.Info("Found AddressSpace for UPS")
+		}
+
+		/*Address space exists and is ready*/
+
+		for _, status := range foundAddressSpace.Status.EndpointStatus {
+			if status.Name == "messaging" { //magic value
+				addressSpaceURL := status.ServiceHost
+				addressConfigMap := newAMQConfigMap(instance, addressSpaceURL)
+				err = r.client.Create(context.TODO(), addressConfigMap)
+				if err != nil && !errors.IsAlreadyExists(err) {
+					return reconcile.Result{}, err
+				}
+
+			}
+		}
+
+		//check that user exists
+		user, err := newMessagingUser(instance)
+
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Set UnifiedPushServer instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, user, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		foundUser := &messaginguserv1beta.MessagingUser{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: user.Name, Namespace: user.Namespace}, foundUser)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("Creating a new MessagingUser", "MessagingUser.Namespace", user.Namespace, "MessagingUser.Name", user.Name)
+			err = r.client.Create(context.TODO(), user)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			secret := newAMQSecret(instance, string(user.Spec.Authentication.Password))
+			err = r.client.Create(context.TODO(), secret)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{RequeueAfter: time.Second * 1}, nil
```

Comment:
At this point, the MessagingUser (and Secret) have been successfully created. Is it necessary to return-and-requeue here, or should it continue straight away to the queues stuff? If a return-and-requeue is needed, is it necessary to wait a second (rather than `RequeueAfter: time.Second * 1`, you could have `Requeue: true`).